### PR TITLE
KCRO-206 : As a developer, I want an API for crud operations on a list of from addresses so that can DKIM to sign emails

### PR DESCRIPTION
**Pull request by manushiperera**
Previously, DKIM information was common for the account and unable to configure based on the available email addresses of the account. Now, private key file and DKIM selector information can be configured for each email address through new `whitelabel` API. A new crud operations are defined to configure, verify and update email addresses along with DKIM information.

Updated properly test cases for email address related crud operations.

### CODE REVIEW

**Comment by manushiperera** [comment]
Related PRs
https://github.com/2600hz/kazoo-core/pull/867
https://github.com/2600hz/kazoo-teletype/pull/73
https://github.com/2600hz/kazoo-crossbar/pull/580

File: src/api/pqc_cb_whitelabel.erl [hunk 1]
```diff
@@ -55,46 +56,65 @@ update_whitelabel(API, AccountId, WhitelabelJObj) ->
...
     pqc_cb_crud:delete(API, whitelabel_url(API, AccountId)).
 
--spec summary_dkim(pqc_cb_api:state(), kz_term:ne_binary()) -> pqc_cb_api:response().
-summary_dkim(API, AccountId) ->
-    pqc_cb_crud:summary(API, dkim_url(API, AccountId)).
+-spec summary_email(pqc_cb_api:state(), kz_term:ne_binary()) -> pqc_cb_api:response().
+summary_email(API, AccountId) ->
+    pqc_cb_crud:summary(API, emails_url(API, AccountId)).
+
+-spec create_email(pqc_cb_api:state(), kz_term:ne_binary(), iodata()) -> pqc_cb_api:response().
```
Comment (jamesaimonetti) [inline src/api/pqc_cb_whitelabel.erl]: 3rd arg is not `iodata()`

File: src/api/pqc_cb_whitelabel.erl [hunk 2]
```diff
@@ -55,46 +56,65 @@ update_whitelabel(API, AccountId, WhitelabelJObj) ->
...
+
+-spec update_email(pqc_cb_api:state(), kz_term:ne_binary(), kz_term:ne_binary(), kz_term:api_object())
+                  -> pqc_cb_api:response().
+update_email(API, AccountId, EmailId, EmailJObj) ->
+    Envelope = pqc_cb_api:create_envelope(EmailJObj),
+    pqc_cb_crud:update(API, email_url(API, AccountId, EmailId), Envelope).
 
--spec create_dkim(pqc_cb_api:state(), kz_term:ne_binary(), iodata()) -> pqc_cb_api:response().
-create_dkim(API, AccountId, Data) ->
+-spec update_dkim(pqc_cb_api:state(), kz_term:ne_binary(), kz_term:ne_binary(), kz_term:api_object())
```
Comment (jamesaimonetti) [inline src/api/pqc_cb_whitelabel.erl]: 4th param is not api_object but `iodata()` i think?

File: src/api/pqc_cb_whitelabel.erl (hunk 1 above)
Comment (manushiperera) [reply src/api/pqc_cb_whitelabel.erl]: Yes James, my bad
It seems I mixed up the data types in the create and update functions.
Thanks for looking into it.

File: src/api/pqc_cb_whitelabel.erl (hunk 2 above)
Comment (manushiperera) [reply src/api/pqc_cb_whitelabel.erl]: Same issue
Thanks James

**Pull request merged**